src/lib/analytics/__tests__/segmentOneTrustIntegration.jest.ts

```diff
@@ -0,0 +1,315 @@
+    const result = await SOI.getConsent()
+    expect(result).toBe("C0001")
+  })
+  it("returns onetrust consent if it gets onetrust consent", async () => {
+    SOI.getOneTrustConsent = jest.fn(() => Promise.resolve("C0001,C0002"))
+    const result = await SOI.getConsent()
+    expect(result).toBe("C0001,C0002")
+  })
+})
+
+describe("getOneTrustConsent", () => {
+  beforeEach(() => {
+    SOI.delay = jest.fn(() => Promise.resolve())
+  })
+  afterAll(() => {
+    SOI.oneTrustReady = originalOneTrustReady
+    SOI.delay = originalDelay
+  })
+  it("returns empty string if onetrust is never ready", async () => {
+    SOI.oneTrustReady = jest.fn(() => {
+      return false
+    })
+    const result = await SOI.getOneTrustConsent()
+    expect(result).toBe("")
+  })
+  it("returns onetrust consent string if onetrust is ready", async () => {
+    SOI.oneTrustReady = jest.fn(() => {
+      return true
+    })
+    window.OnetrustActiveGroups = "C0001"
+    const result = await SOI.getOneTrustConsent()
+    expect(result).toBe("C0001")
+  })
+})
+
+describe("oneTrustReady", () => {
+  it("returns true if onetrust consent string contains C0001", () => {
+    window.OnetrustActiveGroups = "C0001"
+    const result = SOI.oneTrustReady()
+    expect(result).toBe(true)
+  })
+  it("returns false if onetrust consent string is empty", () => {
+    window.OnetrustActiveGroups = ""
+    const result = SOI.oneTrustReady()
+    expect(result).toBe(false)
+  })
+  it("returns false if onetrust consent string is non-empty but does not contain C0001", () => {
+    window.OnetrustActiveGroups = ",,"
+    const result = SOI.oneTrustReady()
+    expect(result).toBe(false)
+  })
+})
+
+describe("fetchDestinations", () => {
+  afterAll(() => {
+    SOI.fetchDestinationForWriteKey = originalFetchDestinationForWriteKey
+  })
+  it("returns destinations", async () => {
+    SOI.fetchDestinationForWriteKey = jest.fn(() =>
+      Promise.resolve([{ id: "foo" }])
+    )
+    const result = await SOI.fetchDestinations("abc")
+    expect(result).toEqual([{ id: "foo" }])
+  })
+  it("removes repeater destinations", async () => {
+    SOI.fetchDestinationForWriteKey = jest.fn(() =>
+      Promise.resolve([{ id: "foo" }, { id: "Repeater" }])
+    )
+    const result = await SOI.fetchDestinations("abc")
+    expect(result).toEqual([{ id: "foo" }])
+  })
+  it("removes duplicate destinations", async () => {
+    SOI.fetchDestinationForWriteKey = jest.fn(() =>
+      Promise.resolve([{ id: "foo" }, { id: "foo" }])
+    )
+    const result = await SOI.fetchDestinations("abc")
+    expect(result).toEqual([{ id: "foo" }])
+  })
+})
+
+describe("fetchDestinationForWriteKey", () => {
+  afterAll(() => {
+    SOI.validateSegmentResponse = originalValidateSegmentResponse
+  })
+  it("returns destinations if segment api returns them", async () => {
+    const res = {
+      ok: true,
+      json: () => Promise.resolve([{ creationName: "foo" }]),
+    }
+    // @ts-ignore
```

Comment:
to fix all of these `ts-ignores` can just drop 

```
declare const window: any
``` 
before the tests start

Reply:
done. thanks!